pipe.go

```diff
@@ -137,33 +138,60 @@ func (s pipeAddress) String() string {
 	return string(s)
 }
 
+//helper function used to try to open the pipe multiple times.
+func tryDialPipe(ctx context.Context, path *string) (syscall.Handle, error) {
```

Comment:
NIT: `// tryDialPipe attempts to dial the pipe at `path` until `ctx` cancellation or timeout.